telegram/ext/_baseupdateprocessor.py

```diff
@@ -0,0 +1,167 @@
+
+        .. seealso::
+            :meth:`shutdown`
+        """
+
+    @abstractmethod
+    async def shutdown(self) -> None:
+        """Shutdown the processor so resources can be freed. Must be implemented by a subclass.
+
+        .. seealso::
+            :meth:`initialize`
+        """
+
+    async def process_update(
+        self,
+        update: object,
+        coroutine: "Awaitable[Any]",
+        application: "Application",
+    ) -> None:
+        """Calls :meth:`do_process_update` with a semaphore to limit the number of concurrent
+        updates.
+
+        Args:
+            update (:obj:`object`): The update to be processed.
+            coroutine (:obj:`Awaitable`): The coroutine that will be awaited to process the update.
+            application (:class:`telegram.ext.Application`): The application instance.
+        """
+        async with self._semaphore:
+            await self.do_process_update(update, coroutine)
+            application.update_queue.task_done()
+
+    async def __aenter__(self) -> "BaseUpdateProcessor":
+        """Simple context manager which initializes the Processor."""
+        try:
+            await self.initialize()
+            return self
+        except Exception as exc:
+            await self.shutdown()
+            raise exc
+
+    async def __aexit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_val: Optional[BaseException],
+        exc_tb: Optional[TracebackType],
+    ) -> None:
+        """Shutdown the Processor from the context manager."""
+        await self.shutdown()
+
+
+class SimpleUpdateProcessor(BaseUpdateProcessor):
+    """Instance of :class:`telegram.ext.BaseUpdateProcessor` that immediately awaits the
+    coroutine, i.e. does not apply any additional processing. This is used by default when
+    :attr:`telegram.ext.ApplicationBuilder.concurrent_updates` is :obj:`int`.
+
+    Objects of this class are comparable in terms of equality. Two objects of this class are
+    considered equal, if their :paramref:`max_concurrent_updates` is equal.
+    """
+
+    __slots__ = ()
+
+    async def do_process_update(
+        self,
+        update: object,
+        coroutine: "Awaitable[Any]",
+    ) -> None:
+        """Immediately awaits the coroutine, i.e. does not apply any additional processing.
+
+        Args:
+            update (:obj:`object`): The update to be processed.
+            coroutine (:obj:`Awaitable`): The coroutine that will be awaited to process the update.
+        """
+        await coroutine
+
+    async def initialize(self) -> None:
+        """Does nothing."""
+
+    async def shutdown(self) -> None:
+        """Does nothing."""
+
+    def __eq__(self, other: object) -> bool:
+        return (
+            isinstance(other, self.__class__)
+            and other.max_concurrent_updates == self.max_concurrent_updates
+        )
+
+    def __hash__(self) -> int:
+        return hash(self.max_concurrent_updates)
```

Comment:
Well, I did that just for tests most probably here https://github.com/python-telegram-bot/python-telegram-bot/pull/3654/files#diff-8708d60b1d1959712bde5982c7f5d73ea53e7d6c522be904062924f574d3a03aR417

Reply:
I see. Adding functionality "just" to make testing easier would be an antipattern IMO :D In the tests you can instead use `is` instead of `==`. That even makes sure that `app.update_processor` is actually the very same object as `expected` and not a different object that just happens to have the same `max_concurrent_updates` setting.

Reply:
Well I tried "is" prior to "==" and it didnt worked, so I decided to check if it is the instance of SUP and max concurrent updates is same, cant find a way to get the same attribute passed tho...